src/main/java/de/tum/in/www1/artemis/web/rest/AtheneResource.java

```diff
@@ -0,0 +1,171 @@
+            TextBlock newBlock = new TextBlock();
+            newBlock.setId(t.id);
+            newBlock.setText(t.text);
+            newBlock.setStartIndex(t.startIndex);
+            newBlock.setEndIndex(t.endIndex);
+            newBlock.automatic();
+
+            // take the corresponding TextSubmission and add the text blocks.
+            // The addBlocks method also sets the submission in the textBlock
+            TextSubmission s = submissionsMap.get(t.submissionId);
+            submissionsMap.get(t.submissionId).addBlock(newBlock);
+            textBlocks.add(newBlock);
+        }
+
+        return textBlocks;
+
+    }
+
+    /**
+     * Process clusters and save to database
+     *
+     * @param clusterMap The map of clusters to process
+     * @param textBlockMap The map of textBlocks belonging to the clusters
+     * @param exerciseId The exerciseId of the exercise the blocks belong to
+     */
+    private void processClusters(Map<Integer, TextCluster> clusterMap, Map<String, TextBlock> textBlockMap, Long exerciseId) {
+        // Remove Cluster with Key "-1" as it is only contains the blocks belonging to no cluster.
+        clusterMap.remove(-1);
+        final List<TextCluster> savedClusters = textClusterRepository.saveAll(clusterMap.values());
+
+        // Find exercise, which the clusters belong to
+        Optional<TextExercise> optionalTextExercise = textExerciseRepository.findWithEagerTeamAssignmentConfigAndCategoriesById(exerciseId);
+        if (optionalTextExercise.isEmpty()) {
+            log.error("Error while processing Athene clusters. Exercise with id " + exerciseId + "not found");
+            new Error().printStackTrace();
+            return;
+        }
+        TextExercise textExercise = optionalTextExercise.get();
+
+        // Link clusters with blocks
+        for (TextCluster cluster : savedClusters) {
+            cluster.setExercise(textExercise);
+            List<TextBlock> updatedBlockReferences = cluster.getBlocks().parallelStream().map(block -> textBlockMap.get(block.getId())).peek(block -> block.setCluster(cluster))
+                .collect(toList());
+            textAssessmentQueueService.setAddedDistances(updatedBlockReferences, cluster);
+            updatedBlockReferences = textBlockRepository.saveAll(updatedBlockReferences);
+            cluster.setBlocks(updatedBlockReferences);
+        }
+
+        // Save clusters in Database
+        textClusterRepository.saveAll(savedClusters);
+    }
+
+    /**
+     * Saves automatic textAssessments of Athene
+     *
+     * @param exerciseId The exerciseId of the exercise which will be saved
+     * @param requestBody The calculation results containing blocks and clusters
+     * @param auth The secret for authorization
+     * @return 200 Ok if successful or 401 unauthorized if secret is wrong
+     */
+    @PostMapping(value = "/{exerciseId}", consumes = APPLICATION_JSON_VALUE)
+    @Transactional
```

Comment:
please avoid Transactional

Reply:
I think we need Transactional here since we write to the database